### PR TITLE
17879 - Fix isBusinessAffiliated check and missing corpNum

### PR DESCRIPTION
**Pull request by seeker25**
https://github.com/bcgov/entity/issues/17879

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

File: auth-web/src/composables/affiliations-factory.ts [hunk 1]
```diff
@@ -43,6 +43,9 @@ export const useAffiliations = () => {
   }
 
   const isBusinessAffiliated = (businessIdentifier: string): boolean => {
+    if (businessIdentifier === undefined) {
+      return false
+    }
```
Comment (severinbeauvais) [inline auth-web/src/composables/affiliations-factory.ts:48]: I would do `if (!businessIdentifier)` to handle the null case, too. (I think we never have id=0.)

Walk me through this. This functions returns false, meaning the business is not affiliated, meaning...? How did the user get here and what happens now?

Comment (seeker25) [reply auth-web/src/composables/affiliations-factory.ts:48]: I think it has to do with another change that was missing - the corpNum wasn't being brought up

File: auth-web/src/stores/business.ts [hunk 2]
```diff
@@ -123,7 +123,8 @@ export const useBusinessStore = defineStore('business', () => {
       requestActionCd: nr.requestActionCd,
       natureOfBusiness: nr.natureBusinessInfo,
       expirationDate: nr.expirationDate,
-      applicants: nr.applicants
+      applicants: nr.applicants,
+      corpNum: nr.corpNum
```
Comment (severinbeauvais) [inline auth-web/src/stores/business.ts:127]: I think there are a few interfaces that should be updated. I think Hongjing did something similar about a month ago with a new property from Namex API.

Comment (seeker25) [reply auth-web/src/stores/business.ts:127]: Let me look at her changes and make sure I didn't miss them refactoring part of this

File: auth-web/src/composables/affiliations-factory.ts (hunk 1 above)
Comment (seeker25) [reply auth-web/src/composables/affiliations-factory.ts:48]: I think it has to do with another change that was missing - the corpNum wasn't being brought up

File: auth-web/src/stores/business.ts (hunk 2 above)
Comment (seeker25) [reply auth-web/src/stores/business.ts:127]: corpNum was missing, that's it I think

**Review by severinbeauvais — APPROVED** [review]

**Pull request merged**